### PR TITLE
feat(nestjs): Support v10

### PR DESCRIPTION
**Pull request by underfisk**
- upgrades our nestjs packages to support `10.x.x`
- upgrades typescript to `4.9.5` which should meet the new requirements
- drops support for node 14

### CODE REVIEW

**Comment by underfisk** [comment]
@WonderPanda This and a few dependencies that I've been upgrading I'd recommend having major releases, even though there are no breaking changes that are noticeable we're bumping major dependencies such as nestjs

**Comment by asselin** [comment]
@underfisk With this change, the lib will ONLY be compatible with Nest 10.  If you try to install it in a project that is on Nest 9, you'll wind up with a copy of Nest 9 AND 10.  For my own libs, I've used ^9.0.0 || ^10.0.0` to indicate compatibility with both.  If you'd prefer not to do that, I'd recommend bumping the major version of these libs up, so that current projects on Nest 9 that use them don't accidentally pull down a new version and break.

And-- thank you for this PR.  I'm using the modules and discovery packages, and am interested in the version bump for those.

**Comment by underfisk** [comment]
> @underfisk With this change, the lib will ONLY be compatible with Nest 10. If you try to install it in a project that is on Nest 9, you'll wind up with a copy of Nest 9 AND 10. For my own libs, I've used ^9.0.0 || ^10.0.0` to indicate compatibility with both. If you'd prefer not to do that, I'd recommend bumping the major version of these libs up, so that current projects on Nest 9 that use them don't accidentally pull down a new version and break.
> 
> And-- thank you for this PR. I'm using the modules and discovery packages, and am interested in the version bump for those.

Hi there
My intention is to support only `10.x.x` but I've advised @WonderPanda to do a major bump so we can indicate this. There are a few more dependencies upgrades that are important like supporting Stripe v11 API but he's the one that has permissions to publish

**Comment by jeromeverreault** [comment]
We have been using NEST 10 since last week with success. But since yesterday morning following the release of @golevelup/nestjs-rabbitmq 3.7.0, our project no longer works. I'm waiting for this PR to see if that fixes the problem. Thank you very much for the effort to support NEST 10!

**Comment by underfisk** [comment]
> We have been using NEST 10 since last week with success. But since yesterday morning following the release of @golevelup/nestjs-rabbitmq 3.7.0, our project no longer works. I'm waiting for this PR to see if that fixes the problem. Thank you very much for the effort to support NEST 10!

I'll be following up on any issue that is unsolved, I feel we should push to the latest (of course using a major release) so that people can always run on the most updated (if possible with the least breaking changes) version because most of them bring security vulnerabilities fixes which for me is a very important thing.
Personally, I haven't had too much free time to develop new features/be using the RabbitMQ package but I know that it has potential. 
I'll be looking for any potential input/pr that comes in so that the library can evolve with the community that utilizes it

**Comment by mccuna** [comment]
Hey everybody! First of all, many thanks for this PR. We've been using this package for a few of our microservices and it's proven to be a real gem for RabbitMQ integrations.

This PR bumps a few major versions, but I've noticed that typescript is bumped only to `typescript@^4.9.5`. Would it be possible to bump this to something like `typescript@^4.9.5 | ^5.0.0` too? New NestJs projects created using `npx nest new` start with `typescript@^5.1.3`. Having it support `typescript@^5.0.0` would allow using the package without tweaking any versions for a newly created project.

**Comment by underfisk** [comment]
> Hey everybody! First of all, many thanks for this PR. We've been using this package for a few of our microservices and it's proven to be a real gem for RabbitMQ integrations.
> 
> This PR bumps a few major versions, but I've noticed that typescript is bumped only to `typescript@^4.9.5`. Would it be possible to bump this to something like `typescript@^4.9.5 | ^5.0.0` too? New NestJs projects created using `npx nest new` start with `typescript@^5.1.3`. Having it support `typescript@^5.0.0` would allow using the package without tweaking any versions for a newly created project.

You're right, i think going to typescript v5 should be the right choice with this PR as nest indeed generates using typescript v5. I'll have to review what it might imply on our packages but if it's fine and there are no breaking changes, I'll bump

**Review by marlon-chagas — APPROVED** [review]
:rocket: I really need it!

File: packages/rabbitmq/package.json
Comment (marlon-chagas) [inline packages/rabbitmq/package.json:1]: ps: When create updates in npm package, you must to upgrade too the `version` key.

Comment (underfisk) [reply packages/rabbitmq/package.json:1]: @marlon-chagas That's going to be handled with the tool we use to publish, @WonderPanda will take of that don't worry :)

**Comment by Hareloo** [comment]
@WonderPanda Can you please check this out?

**Comment by phDooY** [comment]
Much anticipated as well. Thanks

**Comment by michelemilani91** [comment]
in the meantime I override the dependency resolution putting this in my `package.json`:
```
"overrides": {
    "@golevelup/nestjs-rabbitmq": {
      "@nestjs/common": "$@nestjs/common",
      "@nestjs/core": "$@nestjs/core"
    }
  }
```

**Comment by underfisk** [comment]
@mccuna Getting back to you with the Typescript version that is required and we're on a good spot. 
You can read more about it https://trilon.io/blog/nestjs-10-is-now-available#cli-plugins-typescript-48 but essentially its `>= 4.8` and we're on `4.9.5` therefore i don't see a reason to risk yet bumping up to v5 in this PR

**Comment by fredx21** [comment]
@WonderPanda this PR needs your approval. We are many holding off nestjs 10 until this is merged.

**Comment by NilsMoller** [comment]
Waiting on this too 😃

**Review by WonderPanda — APPROVED** [review]

**Pull request merged**
